### ISSUE TITLE
Clippy tells me to put `# Error` section on doc comments for `#[doc(hidden)]` function

### ISSUE THREAD

**Issue by Robbepop** [rust-lang/rust-clippy#11501]
### Summary

Clippy tells me to write an `# Error` doc section for a function that has hidden its documentation via `#[doc(hidden)]`. In my opinion this `clippy` warning thus does not make sense since docs are unavailable anyways.

This warning got introduced when I upgraded to the most recent `clippy` nightly. (`clippy 0.1.74 (8142a319 2023-09-13)`)

### Reproducer

I tried this code:

```rust
#![warn(clippy::missing_errors_doc)]

trait Foo {
    #[doc(hidden)]
    fn foo() -> Result<(), ()>;
}
```

I expected to see this happen: OK

Instead, this happened:
```
warning: docs for function returning `Result` missing `# Errors` section
  --> main.rs:5:5
   |
5  |     fn foo() -> Result<(), ()>;
```

### Version

```text
rustc 1.74.0-nightly (8142a319e 2023-09-13)
binary: rustc
commit-hash: 8142a319ed5c1d1f96e5a1881a6546e463b77c8f
commit-date: 2023-09-13
host: aarch64-apple-darwin
release: 1.74.0-nightly
LLVM version: 17.0.0
```

**Comment by Alexendoo** [comment]
Couldn't reproduce this, is that the full file?

**Comment by Robbepop** [comment]
> Couldn't reproduce this, is that the full file?

Oh I am extremely sorry! I assumed too much about this particular error. The code I pasted is actually not reproducing for me either. I just tried to come up with a minimal code sample of the offending code but I failed so far. The code that triggers this bug can be found here: https://github.com/paritytech/wasmi/blob/837bfdb7a0fc668b64a00ef56dd09187ef2ce7b8/crates/wasmi/src/func/into_func.rs#L98

**Comment by Alexendoo** [comment]
Thanks! That reproduces for me as well

**Comment by Alexendoo** [comment]
The reproducer in the original post needed just one extra thing for it to lint - for it to be `pub trait Foo { .. }`

**Comment by Robbepop** [comment]
> The reproducer in the original post needed just one extra thing for it to lint - for it to be `pub trait Foo { .. }`

yeah that actually makes sense since otherwise it won't be included in the public docs of a crate anyways. 🤦
